modules/GenericDataParser/src/server/generic_data_parser.R

```diff
@@ -2908,9 +2908,330 @@ splitOnSemicolon <- function(x) {
+                      if(!missingParameters) {
+                        warnings[[length(warnings)+1]] <- paste0("The ", statName, " for curve id '", name, "' is ", statValueText, " which is ", threshold$operator, " than the threshold value of ", threshold$value, ".")
+                        title <- paste0(statName," value of ",statValueText, " is ", threshold$operator, " than threshold value ", threshold$value)
+                      }
+                    } else if(class == "error") {
+                      tdStyleClass <- "error"
+                      trStyleClass <- "error"
+                      if(!missingParameters) {
+                        errors[[length(errors)+1]] <- paste0("The ", statName, " for curve id '", name, "' is ", statValueText, " which is ", threshold$operator, " than the threshold value of ", threshold$value, ".")
+                        title <- paste0(statName," value of ",statValueText, " is ", threshold$operator, " than threshold value ", threshold$value)
+                      }
+                    }
+                    if(tdStyleClass == "warning" && trStyleClass == "") {
+                      trStyleClass <- "warning"
+                    }
+                }
+            }
+            if(length(errors) > 0) {
+              for(e in errors) {
+                addError(e)
+              }
+            } else if(length(warnings) > 0) {
+              for(w in warnings) {
+                warnUser(w)
+              }
+            }
+            if(title != "") {
+              title <- sprintf('title="%s"',title)
+            }
+            statsCells <- paste0(statsCells,sprintf('<td %s class="stat_%s">%s</td>', title, tdStyleClass, statValueText))
+        }
+        tr <- ''
+        tr <- paste0(tr,sprintf('<td>%s</td>', batchCode))
+        tr <- paste0(tr,sprintf('<td>%s</td>', fitData$name))
+        tr <- paste0(tr,sprintf('<td>%s</td>', trStyleClass))
+        tr <- paste0(tr,sprintf('<td><img src="data:image/png;base64,%s"></td>', imgTxt))
+        tr <- paste0(tr, statsCells)
+        tr <- paste0(sprintf('<tr class="%s">', trStyleClass), tr)
+        tr <- paste0(tr,"</tr>" )
+        list(SSE, SST, SSR, rSquared, trStyleClass, tr)
+        
+    }, by = c("curveId", "batchCode")]
+
+    table <- '<h4>Dose Response Summary</h4><table class="table table-bordered bv_doseResponseSummaryTable" >'
+    thead <- "<thead>"
+    tr <- "<tr>"
+    for(stat in columnNames <- c(mainCode, "curve id", "class", "Curve", "SSE", "SST", "R&#178;")) {
+        style <- ""
+        if(stat == "Curve") {
+          style <- paste0(' style="min-width: ', defaultRenderingParams$width,'px;"')
+        }
+        if(stat == mainCode) {
+          style <- paste0(' style="min-width: 140px;"')
+        }
+        if(stat == "curve id") {
+          style <- paste0(' style="min-width: 60px;"')
+        }
+        tr <- paste0(tr,sprintf('<th%s>%s</th>', style, stat))
+    }
+    thead <- paste0(thead,tr)
```

Comment:
nit, optional: Could we use a library to do string interpolation to make this a little less paste-y?
https://stackoverflow.com/questions/30711019/better-string-interpolation-in-r

Reply:
Still pasty but a little cleaner.